### PR TITLE
HBASE-25467 Fix error prone problems on branch-2.2

### PR DESCRIPTION
**Pull request by petersomogyi**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 47s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  1s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 20s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 47s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 12s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   4m 28s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   6m  0s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 40s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 40s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 26s |  hbase-server: The patch generated 2 new + 1 unchanged - 0 fixed = 3 total (was 1)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 19s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  29m 42s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 53s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 46s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 166m 34s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 10s |  The patch does not generate ASF License warnings.  |
|  |   | 251m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2850 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 51c7607d616c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2850/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 20a9bad7ca |
| Default Java | Oracle Corporation-1.8.0_275-b01 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/1/testReport/ |
| Max. process+thread count | 2914 (vs. ulimit of 12500) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/1/console |
| versions | git=2.20.1 maven=3.6.3 findbugs=3.1.11 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 11s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 23s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   2m  0s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  4s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 25s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 43s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  31m 39s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   1m  8s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 16s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 158m 57s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 13s |  The patch does not generate ASF License warnings.  |
|  |   | 241m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2850 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d34da259d66f 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2850/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 821f422175 |
| Default Java | Oracle Corporation-1.8.0_275-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/2/testReport/ |
| Max. process+thread count | 3553 (vs. ulimit of 12500) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2850/2/console |
| versions | git=2.20.1 maven=3.6.3 findbugs=3.1.11 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java [hunk 1]
```diff
@@ -68,7 +68,7 @@
  *
  */
 @InterfaceAudience.Public
-public interface RegionInfo {
+public interface RegionInfo extends Comparable<RegionInfo> {
```
Comment (virajjasani) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java:71]: Does this impact compatibility? If not, changes look good.

Comment (petersomogyi) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java:71]: No, it shouldn't. This is the same change that went into HBase 2.3.0+ in HBASE-23753.

**Review by virajjasani — APPROVED** [review]
+1

**Pull request merged**